workshops/forms.py

```diff
@@ -33,35 +34,89 @@
     html5_required = True
 
-    def __init__(self, form=None):
+    def __init__(self, form=None,
+                 duplicate_buttons_on_top=False,
+                 submit_label='Submit',
+                 submit_name='submit',
+                 use_get_method=False,
+                 wider_labels=False,
+                 add_submit_button=True,
+                 add_delete_button=False,
+                 additional_form_class='',
+                 form_tag=True,
+                 display_labels=True,
+                 form_action=None):
+        """
+        `duplicate_buttons_on_top` -- Whether submit buttons should be
+        displayed on both top and bottom of the form.
+
+        `use_get_method` -- Force form to use GET instead of default POST.
+
+        `wider_labels` -- SWCEventRequestForm and DCEventRequestForm have
+        long labels, so this flag (set to True) is used to address that issue.
+
+        `add_delete_button` -- displays additional red "delete" button.
+        If you want to use it, you need to include in your template the
+        following code:
+
+            <form action="delete?next={{ request.GET.next|urlencode }}" method="POST" id="delete-form">
+              {% csrf_token %}
+            </form>
+
+        This is necessary, because delete button must be reassigned from the
+        form using this helper to "delete-form". This reassignment is done
+        via HTML5 "form" attribute on the "delete" button.
+
+        `display_labels` -- Set to False, when your form has only submit
+        buttons and you want these buttons to be aligned to left.
+        """
+
         super().__init__(form)
 
         self.attrs['role'] = 'form'
-        self.inputs.append(Submit('submit', 'Submit'))
 
+        self.duplicate_buttons_on_top = duplicate_buttons_on_top
 
-class BootstrapHelperGet(BootstrapHelper):
-    """Force form to use GET instead of default POST."""
-    form_method = 'get'
+        self.submit_label = submit_label
 
+        if use_get_method:
+            self.form_method = 'get'
 
-class BootstrapHelperWithAdd(BootstrapHelper):
-    """Change form's 'Submit' to 'Add'."""
+        if wider_labels:
+            assert display_labels
+            self.label_class = 'col-lg-3'
+            self.field_class = 'col-lg-7'
+        elif display_labels:
+            self.label_class = 'col-lg-2'
+            self.field_class = 'col-lg-8'
+        else:
+            self.label_class = ''
+            self.field_class = 'col-lg-12'
 
-    def __init__(self, form=None):
-        super().__init__(form)
+        if add_submit_button:
+            self.add_input(Submit(submit_name, submit_label))
+
+        if add_delete_button:
+            self.add_input(Submit(
+                'delete', 'Delete',
+                onclick='return confirm("Are you sure you want to delete it?");',
+                form='delete-form',
+                css_class='btn-danger',
+                style='float: right;'))
```

Comment:
While in most I'm okay with refactoring of bootstrap helper classes, this deletion button seems wrong, because we need more customized messages in `onclick` (e.g. "Are you sure you want to remove 2016-07-30-Krakow event?"). It also requires (according to the docstring) a separate POST form while in the previous approach a link was enough.